zcash_client_sqlite/src/wallet/init/migrations/tx_retrieval_queue.rs

```diff
@@ -62,12 +79,44 @@ impl RusqliteMigration for Migration {
             named_params![":default_expiry_delta": DEFAULT_TX_EXPIRY_DELTA],
         )?;
 
+        // Call `decrypt_and_store_transaction` for each transaction known to the wallet to
+        // populate the enhancement queues with any transparent history information that we don't
+        // already have.
+        let mut stmt_transactions =
+            transaction.prepare("SELECT raw, mined_height FROM transactions")?;
+        let mut rows = stmt_transactions.query([])?;
+        while let Some(row) = rows.next()? {
+            let tx_data = row.get::<_, Option<Vec<u8>>>(0)?;
+            let mined_height = row.get::<_, Option<u32>>(1)?.map(BlockHeight::from);
+
+            if let Some(tx_data) = tx_data {
+                let tx = zcash_primitives::transaction::Transaction::read(
+                    &tx_data[..],
+                    // We assume unmined transactions are created with the current consensus branch ID.
+                    mined_height
+                        .map_or(BranchId::Sapling, |h| BranchId::for_height(&self.params, h)),
+                )
+                .map_err(|_| {
+                    WalletMigrationError::CorruptedData(
+                        "Could not read serialized transaction data.".to_owned(),
+                    )
+                })?;
+
+                wallet::store_decrypted_tx(
+                    transaction,
+                    &self.params,
+                    DecryptedTransaction::new(mined_height, &tx, vec![], vec![]),
+                )?;
```

Comment:
Hmm. This is a very complicated function to be calling in a migration. If we need to change it, will we duplicate it like we do for code called in other migrations?

Reply:
What I would like to do with `store_decrypted_tx` in general is to refactor it to abstract over the database operations (i.e. factor out the business logic) and perhaps even move the main logic to `zcash_client_backend` (relying upon some lower-level data access API trait.) That will make it easier to modularize it in a way that makes the migration maintenance less annoying.